src/ngx_pagespeed.cc

```diff
@@ -422,6 +422,20 @@ ngx_int_t ps_set_cache_control(ngx_http_request_t* r, char* cache_control) {
   return NGX_OK;
 }
 
+StringPiece ps_get_cache_control(ngx_http_request_t* r) {
+  ngx_table_elt_t* header;
+  NgxListIterator it(&(r->headers_out.headers.part));
+  while ((header = it.Next()) != NULL) {
+    if (STR_CASE_EQ_LITERAL(header->key, "Cache-Control")) {
```

Comment:
My understanding is that when headers are written initially they're both added as a header and parsed into `ngx_http_headers_out_t.cache_control`, and after that point anyone who modifies headers is supposed to change both places.  For example, changing both places is what our `ps_set_cache_control` above does, and what `ngx_http_add_cache_control` does. So the literal header is what's sent out, and the parsed version is for the convenience of anyone who wants to read the current headers without parsing them anew.

So it's safe for us to read either `r->headers_out.headers` and find the `Cache-Control` one, or to reassemble a string from `r->headers_out.cache_control`.

@PiotrSikora , does this comment sound right to you?